### PR TITLE
Cisco umbrella investigate parameterized threshold

### PR DESCRIPTION
**Pull request by mmhw**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3914)

## Description
Added the **Suspicious threshold** parameter for better management the DBot Score.
Added a function to verify the Suspicious threshold and Malicious threshold.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate_test.py [hunk 1]
```diff
@@ -8,6 +8,34 @@
...
+@pytest.mark.parametrize('suspicous, malicious, expected_mock_result', [
+    (0, -100, None),
+    (0, -200, ERROR_VERIFY_THRESHOLD_MESSAGE),
+    (200, -100, ERROR_VERIFY_THRESHOLD_MESSAGE),
+    (0, 50, ERROR_VERIFY_THRESHOLD_MESSAGE)
+])
+def test_verify_threshold_suspicouns_and_malicious_parameters(suspicous, malicious, expected_mock_result, mocker):
+    """
+        Given:
+            - The suspicouns and malicious params
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate_test.py:24]:
```suggestion
            - The suspicious and malicious thresholds params
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate_test.py [hunk 2]
```diff
@@ -8,6 +8,34 @@
...
+        When:
+            - Running the integration
+        Then:
+            - Verify suspicouns is bigger then malicious and both of them in range of -100 to 100
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate_test.py:28]:
```suggestion
            - Verify suspicious is bigger then malicious and both of them in range of -100 to 100
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py [hunk 3]
```diff
@@ -105,6 +106,14 @@
 ''' HELPER FUNCTIONS '''
 
 
+def verify_threshold_params(suspicions_threshold, malicious_threshold):
+    if not (100 >= suspicions_threshold > malicious_threshold >= -100):  # type: ignore
+        return_error(
+            "Please provide valid values for the Suspicouns and Malicious Thresholds, when Suspicouns is greater than "
+            "Malicious and both of them are in range of -100 to 100"
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py:113]:
```suggestion
            "Please provide valid threshold values for the Suspicious and Malicious thresholds when Suspicious is greater than Malicious and both are within a range of -100 to 100"
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml [hunk 4]
```diff
@@ -34,10 +34,17 @@ configuration:
   name: baseURL
   required: true
   type: 0
+- defaultvalue: '0'
+  display: DBot Score Suspicous Threshold (-100 to 100)
+  name: suspicous_threshold
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml:39]:
```suggestion
  name: suspicious_threshold
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml [hunk 5]
```diff
@@ -34,10 +34,17 @@ configuration:
   name: baseURL
   required: true
   type: 0
+- defaultvalue: '0'
+  display: DBot Score Suspicous Threshold (-100 to 100)
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml:38]:
```suggestion
  display: DBot Score Suspicious Threshold (-100 to 100)
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml [hunk 6]
```diff
@@ -34,10 +34,17 @@ configuration:
...
+  name: suspicous_threshold
+  required: false
+  additionalinfo: Make sure Suspicous threshold is bigger than Malicious threshold
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml:41]:
```suggestion
  additionalinfo: Make sure Suspicious threshold is bigger than Malicious threshold
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml [hunk 7]
```diff
@@ -34,10 +34,17 @@ configuration:
...
+  type: 0  
 - defaultvalue: '-100'
   display: DBot Score Malicious Threshold (-100 to 100)
   name: dboscore_threshold
   required: false
+  additionalinfo: Make sure Malicious threshold is less than Suspicous threshold
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.yml:47]:
```suggestion
  additionalinfo: Make sure Malicious threshold is less than Suspicious threshold
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py [hunk 8]
```diff
@@ -184,9 +193,9 @@ def securerank_to_dbotscore(sr):
     # converts cisco umbrella score to dbotscore
     DBotScore = 0
     if sr is not None:
-        if sr > 0 and sr <= 100:
+        if SUSPICOUS_THRESHOLD < sr <= 100:
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py:196]:
```suggestion
        if SUSPICIOUS_THRESHOLD < sr <= 100:
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py [hunk 9]
```diff
@@ -184,9 +193,9 @@ def securerank_to_dbotscore(sr):
...
             DBotScore = 1
-        elif sr < 0 and sr > MALICIOUS_THRESHOLD:
+        elif MALICIOUS_THRESHOLD < sr <= SUSPICOUS_THRESHOLD:
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py:198]:
```suggestion
        elif MALICIOUS_THRESHOLD < sr <= SUSPICIOUS_THRESHOLD:
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py [hunk 10]
```diff
@@ -105,6 +106,14 @@
 ''' HELPER FUNCTIONS '''
 
 
+def verify_threshold_params(suspicions_threshold, malicious_threshold):
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py:109]:
```suggestion
def verify_threshold_params(suspicious_threshold, malicious_threshold):
```

File: Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py [hunk 11]
```diff
@@ -105,6 +106,14 @@
...
+    if not (100 >= suspicions_threshold > malicious_threshold >= -100):  # type: ignore
```
Comment (bziser) [inline Packs/Cisco-umbrella/Integrations/Cisco-umbrella-investigate/Cisco-umbrella-investigate.py:110]:
```suggestion
    if not (100 >= suspicious_threshold > malicious_threshold >= -100):  # type: ignore
```

**Review by bziser — COMMENTED** [review]
Great job! 🤗 

Please fix the misspelling of the "suspicious" word in the code. 
Add global variables for the "100" and "-100" parameters so that in the future someone can easily change the values.

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.

**Review by bziser — APPROVED** [review]

**Review by bziser — APPROVED** [review]

**Pull request merged**